modules/nvidia_plugin/src/cuda_infer_request.cpp

```diff
@@ -298,427 +161,74 @@ void CudaInferRequest::startPipeline(const ThreadContext& threadContext) {
-            devBlob->allocate();
-            _deviceInputs[name] = devBlob;
-        }
-        const bool preProcRequired = preProcessingRequired(foundInput, userBlob, devBlob);
-        if (compoundBlobPassed && !preProcRequired) {
-            IE_THROW(NotImplemented) << "cannot set compound blob: supported only for input pre-processing";
-        }
-
-        if (preProcRequired) {
-            addInputPreProcessingFor(name, userBlob, devBlob ? devBlob : _inputs[name]);
-        } else {
-            size_t inputSize = devBlob->getTensorDesc().getLayout() != InferenceEngine::Layout::SCALAR
-                                   ? InferenceEngine::details::product(devBlob->getTensorDesc().getDims())
-                                   : 1;
-            if (dataSize != inputSize) {
-                IE_THROW() << "Input blob size is not equal network input size (" << dataSize << "!=" << inputSize
-                           << ").";
-            }
-            _inputs[name] = userBlob;
-            devBlob = userBlob;
-        }
-        _batched_inputs.erase(name);
-    } else {
-        if (compoundBlobPassed) {
-            IE_THROW(NotImplemented) << "cannot set compound blob: supported only for input pre-processing";
-        }
-        auto& devBlob = network_output_blobs_[name];
-        auto usrDims = userBlob->getTensorDesc().getDims();
-        auto usrLayout = userBlob->getTensorDesc().getLayout();
-        auto devDims = devBlob->getTensorDesc().getDims();
-        auto devLayout = devBlob->getTensorDesc().getLayout();
-        auto devPrecision = devBlob->getTensorDesc().getPrecision();
-        if (output_dynamic && (devDims != usrDims || devLayout != usrLayout)) {
-            devBlob = make_blob_with_precision({devPrecision, usrDims, TensorDesc::getLayoutByDims(usrDims)});
-            devBlob->allocate();
-            network_output_blobs_[name] = devBlob;
-        }
-        size_t outputSize = devBlob->getTensorDesc().getLayout() != InferenceEngine::Layout::SCALAR
-                                ? details::product(devBlob->getTensorDesc().getDims())
-                                : 1;
-        if (dataSize != outputSize) {
-            IE_THROW() << "Output blob size is not equal network output size (" << dataSize << "!=" << outputSize
-                       << ").";
-        }
-        if (foundOutput->getPrecision() != userBlob->getTensorDesc().getPrecision()) {
-            IE_THROW(ParameterMismatch)
-                << "Failed to set Blob with precision not corresponding to user output precision";
-        }
-        _outputs[name] = userBlob;
-    }
+void CudaInferRequest::cancel() {
+    cancellation_token_.cancel();
+    get_nvidia_model()->get_memory_pool()->Interrupt();
 }
 
-void CudaInferRequest::SetBlobsImpl(const std::string& name, const InferenceEngine::BatchedBlob::Ptr& batchedBlob) {
-    _batched_inputs[name] = batchedBlob;
+void CudaInferRequest::infer() {
+    OPENVINO_NOT_IMPLEMENTED;
```

Comment:
Cuda doesn't support syncronious execution?

Reply:
> Cuda doesn't support syncronious execution?

Synchronous API is working via asynchronous